app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/search/util/SearchTypeDataExtensions.kt

```diff
@@ -102,15 +102,18 @@ private fun CipherView.filterBySearchType(
     searchTypeData: SearchTypeData.Vault,
 ): Boolean =
     when (searchTypeData) {
-        SearchTypeData.Vault.All -> true
```

Comment:
Instead of adding `&& deletedDate == null` to all of these, could we instead do that as some kind of overall check?

Reply:
I was gonna do that but I need the trash case to be different

Reply:
OK. Well there's probably a way to do this like

```kotlin
    when {
        searchTypeData is SearchTypeData.Vault.Trash -> deletedDate != null
        deletedDate != null -> false
        else -> {
            when (searchTypeData) {
                ...
            }
        }
    }
```

but that has some downsides too. What you have is fine.